### PR TITLE
Added SubmissionRequest message type to replace three others

### LINKED ISSUE #625: Harmonize SubmissionRequest and SubmissionsForCourseRequest
I seems too difficult to harmonize these two RPCs because of the return types:
```proto
    rpc GetSubmissions(SubmissionRequest) returns (Submissions) {}
    rpc GetSubmissionsByCourse(SubmissionsForCourseRequest) returns (CourseSubmissions) {}
```
However, we can use the same request type for both RPC methods:
```proto 
message SubmissionRequest {
    uint64 courseID = 1;
    enum SubmissionType {
        ALL   = 0;  // fetch all submissions
        USER  = 1;  // fetch all user submissions
        GROUP = 2;  // fetch all group submissions
    }
    oneof fetchMode {
        uint64 userID       = 2;  // fetch single user's submissions
        uint64 groupID      = 3;  // fetch single group's submissions
        uint64 submissionID = 4;  // fetch single specific submission
        SubmissionType type = 5;  // fetch all submissions of given type
    }
    bool withBuildInfo = 6;
}
```
The `withBuildInfo` may only be needed for the **fetch all submissions** mode. Consider whether or not to respect `withBuildInfo` for the single user or group modes.

### PR DESCRIPTION
**Pull request by meling**
The SubmissionRequest message type replaces the SubmissionRequest, SubmissionReviewersRequest, and SubmissionsForCourseRequest message types.

Based on #904.
Fixes #625

### CODE REVIEW

**Comment by JosteinLindhom** [comment]
I made some changes such that our code behaves similarly to before, but with some quirks:

Sending a `GetSubmission` request with `FetchMode` set to anything other than `SubmissionID` will return the latest submission in our database. If the latest submission belongs to a course the caller does not have claims for, the submission will not be returned. Might be worth adding a check to return an error if `FetchMode` is not set to `SubmissionID`.

`GetSubmissions` requests with `UserID` set will only return submissions if the ID matches that in the user claims.
If `GroupID` is set, submissions are only returned for students if the ID matches any group ID in claims. For teachers, submissions are returned for all group IDs. If a group ID is valid, but belongs to another course than the one sent in the request, an empty response is returned.

For `GetSubmissionsByCourse`, if no `Type` is set (or any other `FetchMode` is used other than `Type`), it will default to `ALL`.

**Comment by meling** [comment]
What happens if a teacher asks for a single userid via fetchmode? I'm thinking about the scenario of downloading the build log for one assignment. (I don't think the current front end supports this case yet.)

(On the phone)

**Comment by meling** [comment]
there are some remaining todos.

File: web/interceptor/access_control.go [hunk 1]
```diff
@@ -129,7 +129,11 @@ func (a *AccessControlInterceptor) WrapUnary(next connect.UnaryFunc) connect.Una
 			case student:
 				// GetSubmissions is used to fetch individual and group submissions.
 				// For individual submissions needs an extra check for user ID in request.
-				if method == "GetSubmissions" && req.IDFor("group") == 0 {
+				if method == "GetSubmissions" {
+					if req.IDFor("group") != 0 {
```
[the remark this answers is not in the document]
Comment (meling) [reply web/interceptor/access_control.go:133]: Okay. Thanks for the offline explanation.

**Pull request merged**